engine/src/main/java/org/hibernate/search/engine/backend/document/converter/FromIndexFieldValueConverter.java

```diff
@@ -14,6 +14,11 @@
  */
 public interface FromIndexFieldValueConverter<F, V> {
 
+	/**
+	 * @return The type of the converted value.
+	 */
+	Class<?> getConvertedType();
```

Comment:
That's the issue I talked to you about on HipChat.

I didn't find a way to make a constraint work with generic classes e.g. `ValueWrapperFromIndexFieldValueConverter` for instance.

If you have an idea, it's very welcome.

Reply:
Right... as soon as `V` is a non-raw type, you'll have a hard time laying hands on an instance of `Class<V>`. Which is normal, because Java erases types, so `Class<ValueWrapper<T>>` does not exist, only `Class<ValueWrapper>` (raw type) does.

And believe it or not, you can assign `ValueWrapper` to `ValueWrapper<T>`, but not `Class<ValueWrapper>` to `Class<ValueWrapper<T>>`. Which is a bit strange if you ask me, but that's just the way it is.

In light of this, and since we cannot express the right constraint (which is "you must return the raw type of T"), what would you think of the following instead:

```java
	/**
	 * Check whether converted values can be assigned to the given type.
	 * <p>
	 * This method is generally implemented like this:
	 * {@code return superTypeCandidate.class.isAssignableFrom( TheConvertedType.class )}.
	 * @param superTypeCandidate A candidate type for assignment of converted values.
	 * @return {@code true} if the converted type {@link V} is a subtype of {@code superTypeCandidate}, {@code false} otherwise.
	 */
	boolean isConvertedTypeAssignableTo(Class<?> superTypeCandidate);
```

As a bonus, that would be consistent with `ToIndexFieldValueConverter.isCompatibleWith()`, and would open the door to more complex type matching in the future (implementations could use a `TypePatternMatcher`).

Reply:
Done.